### ISSUE TITLE
DynamoDB Client Throwing Error on New Version

### ISSUE THREAD

**Issue by gakinson** [aws/aws-sdk-js-v3#4237]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

Getting the follow error when doing a simple Scan command using the DynamoDB Document client:

```
TypeError: Cannot read property 'S' of null
    at AttributeValueFilterSensitiveLog (/Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1144:13)
    at /Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1492:65
    at Array.reduce (<anonymous>)
    at /Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1490:61
    at Array.map (<anonymous>)
    at ScanOutputFilterSensitiveLog (/Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1490:26)
    at /Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:16:21

```

### SDK version number

@aws-sdk/client-dynamodb@3.218.0, ...

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v14.18.3

### Reproduction Steps

```
import {DynamoDBClient} from "@aws-sdk/client-dynamodb";
import {DynamoDBDocumentClient, QueryCommand, ScanCommand, UpdateCommand} from "@aws-sdk/lib-dynamodb";

const client = new DynamoDBClient();
const ddbClient = DynamoDBDocumentClient.from(client, {marshallOptions: {removeUndefinedValues: true}});

await ddbClient.send(new ScanCommand({TableName: '<TABLE_NAME>'}))).
```

### Observed Behavior

```
TypeError: Cannot read property 'S' of null
    at AttributeValueFilterSensitiveLog (/Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1328:13)
    at /Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1593:149
    at Array.reduce (<anonymous>)
    at /Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1593:61
    at Array.map (<anonymous>)
    at ScanOutputFilterSensitiveLog (/Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/client-dynamodb/dist-cjs/models/models_0.js:1593:26)
    at /Users/geoffreykinson/IdeaProjects/wesponsored-scripts/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:16:21
    at processTicksAndRejections (internal/process/task_queues.js:95:5)

```

### Expected Behavior

Successfully execute command

### Additional Information/Context

If I revert the dependencies back to 3.188.0, the above command works.

i tried going back a few version (3.216.0, 3.215.0) and still got the same error. It was not until I went back to something further it worked again.

**Comment by adrai** [comment]
probably duplicate of #4222

**Comment by RanVaknin** [comment]
Hi @gakinson ,

This issue is related to this [initial issue](https://github.com/aws/aws-sdk-js-v3/issues/3846). The team is overwhelmed with some re:Invent related release bugs. Once we can get that issue sorted out this issue will get looked at with first priority.

Since there are 2 other issues open, I'm going to close this in favor of tracking it over there. Feel free to comment on that initial issue to say that youre also impacted. 

Apologies for the inconvenience.
Ran~
